Homework/2/Program.cs

```diff
@@ -0,0 +1,470 @@
+                throw;
+            }
+
+        }
+
+        // Function to write user info from userArray to users.txt file
+        static void WriteUsersToFile()
+        {
+            try
+            {
+                using (StreamWriter writer = new StreamWriter(pathToFile,false))
+                {
+                    for (int i = 0; i < userArray.Length; i++)
+                    {
+                        writer.WriteLine($"{userArray[i][0]};{userArray[i][1]}");
+                    }
+                }
+            }
+            catch (Exception defaultEx)
+            {
+                throw;
+            }
+        }
+
+        //Funtion that gets which option was selected and calls other needed functions via switch statement.
+        static void OptionSelection(string option) {
+            Console.Clear();
+            switch (option)
+            {
+                case "1":
+                    LogIn();
+                    break;
+                case "2":
+                    PromptAddUserToArray();
+                    break;
+                case "3":
+                    ListUsers();
+                    break;
+                case "4":
+                    PromptAddUserToArray(addToEnd: false, addToStart: true);
+                    break;
+                case "5":
+                    PromptAddUserToArray();
+                    break;
+                case "6":
+                    PromptAddUserToArray(addToEnd: false, addToStart: false);
+                    break;
+                case "7":
+                    PromptRemoveUser(removeFromEnd: false, removeFromStart: true);
+                    break;
+                case "8":
+                    PromptRemoveUser(removeFromEnd: true, removeFromStart: false);
+                    break;
+                case "9":
+                    PromptRemoveUser(removeFromEnd: false, removeFromStart: false);
+                    break;
+                case "10":
+                    SortUsers();
+                    break;
+                case "11":
+                    LogOut();
+                    break;
+                case "12":
+                    WriteUsersToFile();
+                    Environment.Exit(0);
+                    break;
+                default:
+                    MenuSelection();
+                    break;
+            }
+        }
+
+        static string PromptInput(string message) {
+            Console.WriteLine(message);
+            return Console.ReadLine();
+        }
+
+        //Login function - gets username/password and checks against array
+        static void LogIn() {
+            Console.Clear();
+            string username = PromptInput("Enter username:");
+            string password = PromptInput("Enter password:");
+
+            if (CheckIfUserExists(username))
+            {
+                string[] userDetails = ReturnUserByUsername(username);
+                Console.WriteLine(userDetails[0] + " " + userDetails[1]);
```

Comment:
Hopefully this line was for debugging purposes :).  Don't want to dump `userDetails[1]` to the console before you've even checked their password.